### PR TITLE
PMM-11788 add Postgresql support to DBaaS

### PR DESCRIPTION
**Pull request by recharte**
PMM-11788

Link to the Feature Build: SUBMODULES-3186

If this PR adds or removes or alters one or more API endpoints, please review and add or update the relevant [API documents](https://github.com/percona/pmm/tree/main/docs/api) as well:

- [ ] API Docs updated

If this PR is related to some other PRs in this or other repositories, please provide links to those PRs:

- Links to related pull requests (optional).

### CODE REVIEW

**Review by gen1us2k — COMMENTED** [review]
Thanks for your changes. It looks good so far. I left a couple of comments and I think that the feature needs more unit tests

File: managed/services/dbaas/kubernetes/types.go [hunk 1]
```diff
@@ -35,11 +35,16 @@ const (
...
+	// FIXME when https://jira.percona.com/browse/K8SPG-309 is fixed change the
+	// template to dbaas-%s-pg-secrets
+	postgresqlSecretNameTmpl = "%s-pguser-%s" //nolint:gosec
 	// DatabaseTypePXC is a pxc database
 	DatabaseTypePXC dbaasv1.EngineType = "pxc"
 	// DatabaseTypePSMDB is a psmdb database
 	DatabaseTypePSMDB dbaasv1.EngineType = "psmdb"
-	externalNLB       string             = "external"
+	// DatabaseTypePOSTGRESQL a postgresql database
+	DatabaseTypePOSTGRESQL dbaasv1.EngineType = "postgresql"
```
Comment (recharte) [inline managed/services/dbaas/kubernetes/types.go:46]:
```suggestion
	DatabaseTypePostgresql dbaasv1.EngineType = "postgresql"
```

File: api/managementpb/dbaas/psmdb_clusters.proto [hunk 2]
```diff
@@ -139,7 +139,7 @@ service PSMDBClusters {
   // GetPSMDBClusterCredentials returns a PSMDB cluster credentials by cluster name.
   rpc GetPSMDBClusterCredentials(GetPSMDBClusterCredentialsRequest) returns (GetPSMDBClusterCredentialsResponse) {
     option (google.api.http) = {
-      post: "/v1/management/DBaaS/PSMDBClusters/GetCredentials"
+      post: "/v1/management/DBaaS/PSMDBCluster/GetCredentials"
```
[the remark this answers is not in the document]
Comment (recharte) [reply api/managementpb/dbaas/psmdb_clusters.proto]: reverted the change

File: api/managementpb/dbaas/pxc_clusters.proto [hunk 3]
```diff
@@ -187,7 +187,7 @@ service PXCClusters {
   // GetPXCClusterCredentials returns a PXC cluster credentials by cluster name.
   rpc GetPXCClusterCredentials(GetPXCClusterCredentialsRequest) returns (GetPXCClusterCredentialsResponse) {
     option (google.api.http) = {
-      post: "/v1/management/DBaaS/PXCClusters/GetCredentials"
+      post: "/v1/management/DBaaS/PXCCluster/GetCredentials"
```
[the remark this answers is not in the document]
Comment (recharte) [reply api/managementpb/dbaas/pxc_clusters.proto]: reverted the change

**Review by percona-csalguero — APPROVED** [review]

**Review by gen1us2k — APPROVED** [review]

**Review by percona-csalguero — APPROVED** [review]

**Pull request merged**